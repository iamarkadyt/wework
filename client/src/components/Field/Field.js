import React from 'react'
import DropdownList from 'react-widgets/lib/DropdownList'
import DateTimePicker from 'react-widgets/lib/DateTimePicker'
import Multiselect from 'react-widgets/lib/Multiselect'
import Moment from 'moment'
import momentLocalizer from 'react-widgets-moment'
import './Field.scss'
import { node, string, func, object, bool, number, arrayOf, oneOfType, instanceOf } from 'prop-types'

Moment.locale('en')
momentLocalizer()

const field = ({
    // different fields require different props
    type,
    name, value, onChange, // all except submit and button
    placeholder, error, // all except checkbox, submit and button
    label, // checkbox, submit, button
    list, // list, multiselect
    rows, // textarea
    disabled, // date
    onClick, style, // button, linkButton, textarea
    inline, // container
    containerStyle, // container
    children
}) => {
    let field = null
    const commonProps = { name, id: name, value, onChange }

    switch (type) {
        case "textarea":
            field = <textarea
                {...commonProps}
                placeholder={placeholder || 'Say something...'}
                className="rw-widget-container rw-widget-input"
                style={{
                    padding: '.5rem',
                    fontSize: '1rem',
                    width: '100%',
                    resize: 'none',
                    ...style
                }}
                rows={rows} />
            break
        case "date":
            field = <DateTimePicker
                {...commonProps}
                placeholder={placeholder || 'Pick a date...'}
                time={false}
                format="MMM DD, YYYY"
                editFormat="M/DD/YYYY"
                disabled={disabled} />
            break
        case "list":
            field = <DropdownList
                {...commonProps}
                placeholder={placeholder || "Choose one..."}
                data={list} />
            break
        case "multiselect":
            field = <Multiselect
                {...commonProps}
                placeholder={placeholder || "Enter a few items..."}
                data={list} />
            break
        case "checkbox":
            field = <label htmlFor={name}>
                <input type="checkbox" {...commonProps} />
                &nbsp;&nbsp;{label}
            </label>
            break
        case "submit":
            field = <button
                type="submit"
                className="button-action">
                {label || children}
            </button>
            break
        case "button":
            field = <button
                className="button"
                style={style}
                onClick={onClick}>
                {label || children}
            </button>
            break
        case "linkButton":
            field = <button
                className="link-button"
                style={style}
                onClick={onClick}>
                {label || children}
            </button>
            break
        default:
            field = <input
                {...commonProps}
                placeholder={placeholder}
                className="rw-widget-container rw-widget-input"
                type={type} />
    }

    return (
        <div style={containerStyle}
            className={['Field-container', inline ? 'inline' : ''].join(' ')}>
            {label
                && type !== "checkbox"
                && type !== "submit"
                && type !== "button"
                && type !== "linkButton"
                && <label htmlFor={name}>{label}</label>}
            {field}
            {error
                && type !== "checkbox"
                && type !== "submit"
                && type !== "button"
                && type !== "linkButton"
                && <span className="error">{error}</span>}
        </div>
    )
}

field.propTypes = {
  type: string,
  name: string,
  value: oneOfType([string, arrayOf(string), bool, instanceOf(Date)]),
  onChange: func,
  placeholder: string,
  error: string,
  label: string,
  list: arrayOf(string),
  rows: oneOfType([string, number]),
  disabled: bool,
  onClick: func,
  style: object,
  inline: bool,
  containerStyle: object,
  children: oneOfType([arrayOf(node), node])
}

export default field
