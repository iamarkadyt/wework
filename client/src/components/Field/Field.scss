@import 'variables.scss';
.Field-container {
    &.inline {
        display: inline;
    }
    width: 100%;
    margin: 1rem 0 0 0;
    input[type=checkbox] {
        transform: scale(1.5);
    }
    .button,
    button[type=submit] {
        min-width: 6rem;
        border-radius: .2rem;
        font-size: 1.1rem;
        padding: .2rem 1rem;
        border-color: $color-button-border;
        background: $color-button-backround;
        color: $color-button-font;
        &:active {
            border-color: $color-button-active-border;
            background: $color-button-active-background;
            color: $color-button-active-font;
        }
        &:hover:not(:active) {
            border-color: $color-button-hover-border;
            background: $color-button-hover-backround;
            color: $color-button-hover-font;
        }
    }
    button[type=submit] {
        float: right;
    }
    .button-action {
        background-color: darken(cornflowerblue, 10%);
        border-color: darken(cornflowerblue, 10%);
        background: linear-gradient(to bottom, darken(cornflowerblue, 10%), darken(cornflowerblue, 20%));
        color: white;
        &:active {
            background: linear-gradient(to bottom, darken(cornflowerblue, 25%), darken(cornflowerblue, 15%));
        }
        &:hover:not(:active) {
            background: linear-gradient(to bottom, darken(cornflowerblue, 8%), darken(cornflowerblue, 18%));
        }
    }
    .link-button {
        background: none;
        border: none;
        font-weight: bold;
        font-size: 1.1rem;
        color: $color-font-secondary;
        &:hover {
            color: $color-font-main;
            cursor: pointer;
        }
    }
    input[type=text],
    input[type=password] {
        font-size: 1rem;
        padding: .6rem;
        width: 100%;
    }
    label {
        display: inline-block;
        padding: .2rem;
    }
    textarea {
        background-color: $color-field-background;
        border-color: $color-field-border;
        color: $color-field-font;
    }
    .error {
        display: inline-block;
        color: $color-font-error;
        padding: .2rem;
    }
}