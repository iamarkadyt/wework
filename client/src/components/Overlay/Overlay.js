import React from 'react'
import './Overlay.scss'

class Overlay extends React.Component {
  componentWillMount() {
    document.body.style = "overflow: hidden"
  }

  componentWillUnmount() {
    document.body.style = null
  }

  handleSubmit(e) {
    const { 
      onBackdropClick 
    } = this.props

    if (e.target.id === 'Overlay-container-backdrop') {
      onBackdropClick()
    }
  }

  render() {
    const { 
      children, 
      centered
    } = this.props

    return (
      <div id="Overlay-container-backdrop" className="Overlay-container" onClick={this.handleSubmit}>
        <div className={["Overlay-content", centered ? "Overlay-content-centered" : null].join(' ')}>
          {children}
        </div>
      </div>
    )
  }
}

export default Overlay
