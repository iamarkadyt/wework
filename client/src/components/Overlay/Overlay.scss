@import 'variables.scss';
.Overlay-container {
    position: absolute;
    top: 0;
    left: 0;
    width: 100%;
    min-height: 100vh;
    display: flex;
    justify-content: center;
    align-items: center;
    padding: calc(#{$navigation_height} + 1rem) 0 2rem 0;
    .content {
        z-index: 1;
        background: white;
        padding: 0.3rem;
        border-radius: .3rem;
        box-shadow: $box_shadow_main;
    }
    .backdrop {
        position: fixed;
        top: 0;
        left: 0;
        width: 100vw;
        height: 100vh;
        background-color: rgba(0, 0, 0, 0.5);
    }
}