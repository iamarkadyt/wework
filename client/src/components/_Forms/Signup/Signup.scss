@mixin flexList {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;    
}

.Signup-container {
    width: 100vw;
    height: 100vh;
    @include flexList();
    form {
        @include flexList();
    }
}