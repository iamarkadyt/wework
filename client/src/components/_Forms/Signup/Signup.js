import React from 'react'
import Field from '../../Field/Field'
import { Link, Redirect } from 'react-router-dom'
import { registerUser } from '../../../state/actions/authActions'
import { connect } from 'react-redux'
import './Signup.css'

class Signup extends React.Component {
    state = {
        name: '',
        email: '',
        password: '',
        passwordConfirm: ''
    }

    render() {
        if (this.props.auth.user.token) {
            return <Redirect to='/profile' />
        }

        return <div className="Signup-container">
            <form onSubmit={e => {
                e.preventDefault()
                this.props.registerUser(this.state)
            }}>
                <Field
                    type="text"
                    name="name"
                    label="Your name:"
                    value={this.state.name}
                    onChange={e => this.setState({ name: e.target.value })}
                    placeholder="John Doe"
                    error={this.props.errors.name} />
                <Field
                    type="text"
                    name="email"
                    label="Your email:"
                    value={this.state.email}
                    onChange={e => this.setState({ email: e.target.value })}
                    placeholder="your.name@mail.com"
                    error={this.props.errors.email} />
                <Field
                    type="password"
                    name="password"
                    label="Password:"
                    value={this.state.password}
                    onChange={e => this.setState({ password: e.target.value })}
                    placeholder="Min. 6 characters"
                    error={this.props.errors.password} />
                <Field
                    type="password"
                    name="passwordConfirm"
                    label="Confirm password:"
                    value={this.state.passwordConfirm}
                    onChange={e => this.setState({ passwordConfirm: e.target.value })}
                    placeholder="Type in your password again"
                    error={this.props.errors.passwordConfirm} />
                <br />
                <span>By creating an account, you agree to Conditions of Use and Privacy Notice of this Website</span>
                <Field type="submit" label="Sign Up" />
            </form>
            <span>Already have an account? <Link to="/login">Log In</Link></span>
        </div>
    }
}

const mapStateToProps = state => ({
    errors: state.err,
    auth: state.auth
})

export default connect(mapStateToProps, { registerUser })(Signup)