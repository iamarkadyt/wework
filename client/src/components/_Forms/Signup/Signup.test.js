/**
 * Contract:
 *
 * What my component renders?
 * (snapshot testing)
 * - form containing 4 text fields
 *   1 div with 1 submit field and 1 text span
 *
 * What do I pass to components I render?
 * (props passing)
 * Look at:
 * - Redirect
 * - form
 * - name field
 * - email field
 * - password field
 * - password confirmation field
 * - submit field
 * - Link component
 *
 * What effects do the props I receive have?
 * (props effects)
 * - when certain subfields are defined in [errors] object, respective Fields show errors (tested in props passing)
 * - vice versa is true: no errors, no red texts
 * - if user is authed, visiting /signup would redirect him to /profile
 * - if user isn't authed, visiting /signup would show Signup component
 *
 * How do I use functions?
 * (functions) > (interaction)
 * - n/a, no internal functions, use of others is tested in (interaction)
 *
 * What happens when user interacts with my component?
 * (interaction)
 * - onSubmit must preventDefault(), call all 4 thunk actions and history.push() with correct args
 * - When user types, respective fields update state through setState calls
 * - When user clicks on login <Link> it navigates to /login
 * - Manually visiting /signup while authenticated redirects to /profile
 *
 * What does the state hold?
 * (state) > (props passing)
 * - name field data
 * - email textfield data
 * - password field data
 * - pwd confirmation field data
 *
 * How and When do I update/invalidate state? 
 * (state) > (interaction)
 * - n/a, all of the below is tested in the (interaction) section
 * - When user types, Fields update respective state parts through setState calls
 * - No invalidation occurs throughout the whole lifecycle of the component
 *
 * How state affects other components?
 * (state) > (props passing)
 * - n/a, tested in (props passing)
 * - Updates text in text Fields
 *
 * Any context?
 * (context) > (interaction)
 * - react-router
 *   > history.push()
 *
 * Lifecycle hooks side effects?
 * - N/A
 *
 * Public ref API?
 * - N/A
 *
 */

import React from 'react'
import { shallow } from 'enzyme'
import { _UnconnectedSignup as Signup } from './Signup'
import Field from '../../Field/Field'
import Overlay from '../../Overlay/Overlay'
import cloneDeep from 'lodash.clonedeep'
import Redirect from 'react-router'

describe('Signup', () => {
  let props
  let mountedComponent

  const getMockProps = () => {
    const actions = {
      registerUser: jest.fn(),
      fetchFollowers: jest.fn(),
      fetchUsersProfile: jest.fn(),
      fetchSubscriptions: jest.fn()
    }

    const errors = {}
    const authedUser = { isAuthenticated: false }

    return cloneDeep({
      authedUser,
      errors,
      ...actions
    })
  }

  const comp = () => {
    if (!mountedComponent) {
      mountedComponent = shallow(<Signup {...props} />)
    }
    return mountedComponent
  }

  beforeEach(() => {
    props = getMockProps()
    mountedComponent = undefined
  })

  describe('props passing', () => {
    describe('Redirect component', () => {
      beforeEach(() => {
        props.authedUser.isAuthenticated = true
      })

      it('receives correct set of props', () => {
        const expectedKeys = [ 'to', 'push' ]
        expect(Object.keys(comp().find('Redirect').props())).toStrictEqual(expectedKeys)
      })

      it('receives correct [to] prop', () => {
        expect(comp().find('Redirect').prop('to')).toBe('/profile')
      })

      it('receives correct [push] prop', () => {
        expect(comp().find('Redirect').prop('push')).toBe(false)
      })
    })

    describe('form', () => {
      it('receives correct set of props', () => {
        const expectedKeys = [ 'onSubmit', 'children' ]
        expect(Object.keys(comp().find('form').props())).toStrictEqual(expectedKeys)
      })

      it('receives correct [onSubmit] prop', () => {
        expect(comp().find('form').prop('onSubmit')).toEqual(expect.any(Function))
      })
    })

    describe('name Field', () => {
      it('receives correct set of props', () => {
        const expectedKeys = [
          'type',
          'name',
          'label',
          'value',
          'onChange',
          'placeholder',
          'error'
        ]
        expect(Object.keys(comp().find(Field).find('[name="name"]').props())).toStrictEqual(expectedKeys)
      })

      it('receives correct [type] prop', () => {
        expect(comp().find(Field).find('[name="name"]').prop('type')).toBe('text')
      })

      it('receives correct [label] prop', () => {
        expect(comp().find(Field).find('[name="name"]').prop('label')).toBe('Your name:')
      })

      it('receives state.name for [value] prop if former is defined', () => {
        const state = 'some state'
        comp().setState({ name: state })
        expect(comp().find(Field).find('[name="name"]').prop('value')).toBe(state)
      })

      it('receives undefined for [value] prop if state.name is undefined', () => {
        comp().setState({ name: undefined })
        expect(comp().find(Field).find('[name="name"]').prop('value')).not.toBeDefined()
      })

      it('receives function for [onChange] prop', () => {
        expect(comp().find(Field).find('[name="name"]').prop('onChange')).toEqual(expect.any(Function))
      })

      it('receives correct [placeholder] prop', () => {
        expect(comp().find(Field).find('[name="name"]').prop('placeholder')).toBe('John Doe')
      })

      it('receives props.errors.name for [error] prop if former is defined', () => {
        const msg = 'error occured'
        props.errors.name = msg
        expect(comp().find(Field).find('[name="name"]').prop('error')).toBe(msg)
      })

      it('receives undefined for [error] prop if props.errors.name is undefined', () => {
        delete props.errors.name
        expect(comp().find(Field).find('[name="name"]').prop('error')).not.toBeDefined()
      })
    })

    describe('email Field', () => {
      it('receives correct set of props', () => {
        const expectedKeys = [
          'type',
          'name',
          'label',
          'value',
          'onChange',
          'placeholder',
          'error'
        ]
        expect(Object.keys(comp().find(Field).find('[name="email"]').props())).toStrictEqual(expectedKeys)
      })

      it('receives correct [type] prop', () => {
        expect(comp().find(Field).find('[name="email"]').prop('type')).toBe('text')
      })

      it('receives correct [label] prop', () => {
        expect(comp().find(Field).find('[name="email"]').prop('label')).toBe('Your email:')
      })

      it('receives state.email for [value] prop if former is defined', () => {
        const state = 'some state'
        comp().setState({ email: state })
        expect(comp().find(Field).find('[name="email"]').prop('value')).toBe(state)
      })

      it('receives undefined for [value] prop if state.email is undefined', () => {
        comp().setState({ email: undefined })
        expect(comp().find(Field).find('[name="email"]').prop('value')).not.toBeDefined()
      })

      it('receives function for [onChange] prop', () => {
        expect(comp().find(Field).find('[name="email"]').prop('onChange')).toEqual(expect.any(Function))
      })

      it('receives correct [placeholder] prop', () => {
        expect(comp().find(Field).find('[name="email"]').prop('placeholder')).toBe('your.name@mail.com')
      })

      it('receives props.errors.email for [error] prop if former is defined', () => {
        const msg = 'error occured'
        props.errors.email = msg
        expect(comp().find(Field).find('[name="email"]').prop('error')).toBe(msg)
      })

      it('receives undefined for [error] prop if props.errors.email is undefined', () => {
        delete props.errors.email
        expect(comp().find(Field).find('[name="email"]').prop('error')).not.toBeDefined()
      })
    })

    describe('password Field', () => {
      it('receives correct set of props', () => {
        const expectedKeys = [
          'type',
          'name',
          'label',
          'value',
          'onChange',
          'placeholder',
          'error'
        ]
        expect(Object.keys(comp().find(Field).find('[name="password"]').props())).toStrictEqual(expectedKeys)
      })

      it('receives correct [type] prop', () => {
        expect(comp().find(Field).find('[name="password"]').prop('type')).toBe('password')
      })

      it('receives correct [label] prop', () => {
        expect(comp().find(Field).find('[name="password"]').prop('label')).toBe('Password:')
      })

      it('receives state.password for [value] prop if former is defined', () => {
        const state = 'some state'
        comp().setState({ password: state })
        expect(comp().find(Field).find('[name="password"]').prop('value')).toBe(state)
      })

      it('receives undefined for [value] prop if state.password is undefined', () => {
        comp().setState({ password: undefined })
        expect(comp().find(Field).find('[name="password"]').prop('value')).not.toBeDefined()
      })

      it('receives function for [onChange] prop', () => {
        expect(comp().find(Field).find('[name="password"]').prop('onChange')).toEqual(expect.any(Function))
      })

      it('receives correct [placeholder] prop', () => {
        expect(comp().find(Field).find('[name="password"]').prop('placeholder')).toBe('Min. 6 characters')
      })

      it('receives props.errors.password for [error] prop if former is defined', () => {
        const msg = 'error occured'
        props.errors.password = msg
        expect(comp().find(Field).find('[name="password"]').prop('error')).toBe(msg)
      })

      it('receives undefined for [error] prop if props.errors.password is undefined', () => {
        delete props.errors.password
        expect(comp().find(Field).find('[name="password"]').prop('error')).not.toBeDefined()
      })
    })

    describe('passwordConfirm Field', () => {
      it('receives correct set of props', () => {
        const expectedKeys = [
          'type',
          'name',
          'label',
          'value',
          'onChange',
          'placeholder',
          'error'
        ]
        expect(Object.keys(comp().find(Field).find('[name="passwordConfirm"]').props())).toStrictEqual(expectedKeys)
      })

      it('receives correct [type] prop', () => {
        expect(comp().find(Field).find('[name="passwordConfirm"]').prop('type')).toBe('password')
      })

      it('receives correct [label] prop', () => {
        expect(comp().find(Field).find('[name="passwordConfirm"]').prop('label')).toBe('Confirm password:')
      })

      it('receives state.passwordConfirm for [value] prop if former is defined', () => {
        const state = 'some state'
        comp().setState({ passwordConfirm: state })
        expect(comp().find(Field).find('[name="passwordConfirm"]').prop('value')).toBe(state)
      })

      it('receives undefined for [value] prop if state.passwordConfirm is undefined', () => {
        comp().setState({ passwordConfirm: undefined })
        expect(comp().find(Field).find('[name="passwordConfirm"]').prop('value')).not.toBeDefined()
      })

      it('receives function for [onChange] prop', () => {
        expect(comp().find(Field).find('[name="passwordConfirm"]').prop('onChange')).toEqual(expect.any(Function))
      })

      it('receives correct [placeholder] prop', () => {
        expect(comp().find(Field).find('[name="passwordConfirm"]').prop('placeholder')).toBe('Type in your password again')
      })

      it('receives props.errors.passwordConfirm for [error] prop if former is defined', () => {
        const msg = 'error occured'
        props.errors.passwordConfirm = msg
        expect(comp().find(Field).find('[name="passwordConfirm"]').prop('error')).toBe(msg)
      })

      it('receives undefined for [error] prop if props.errors.passwordConfirm is undefined', () => {
        delete props.errors.passwordConfirm
        expect(comp().find(Field).find('[name="passwordConfirm"]').prop('error')).not.toBeDefined()
      })
    })

    describe('submit Field', () => {
      it('receives correct set of props', () => {
        const expectedKeys = [ 
          'type', 
          'name',
          'inline', 
          'containerStyle', 
          'label'
        ]
        expect(Object.keys(comp().find(Field).find('[name="submit"]').props())).toStrictEqual(expectedKeys)
      })

      it('receives correct [type] prop', () => {
        expect(comp().find(Field).find('[name="submit"]').prop('type')).toBe('submit')
      })

      it('receives correct [inline] prop', () => {
        expect(comp().find(Field).find('[name="submit"]').prop('inline')).toBe(true)
      })

      it('receives correct [containerStyle] prop', () => {
        const expectedStyles = {
          margin: 0
        }
        expect(comp().find(Field).find('[name="submit"]').prop('containerStyle')).toStrictEqual(expectedStyles)
      })

      it('receives correct [label] prop', () => {
        expect(comp().find(Field).find('[name="submit"]').prop('label')).toStrictEqual('Sign Up')
      })
    })

    describe('Link component', () => {
      it('receives correct set of props', () => {
        const expectedKeys = [
          'to',
          'children',
          'replace'
        ]
        expect(Object.keys(comp().find('span.Signup-options-info').find('Link').props())).toStrictEqual(expectedKeys)
      })

      it('receives correct [to] prop', () => {
        expect(comp().find('span.Signup-options-info').find('Link').prop('to')).toBe('/login')
      })

      it('receives correct [children] prop', () => {
        expect(comp().find('span.Signup-options-info').find('Link').prop('children')).toBe('Log In')
      })
    })
  })

  describe('props effects', () => {
    describe('authentication status', () => {
      it('renders only Redirect if user is authed', () => {
        props.authedUser.isAuthenticated = true
        expect(comp().first().name()).toBe('Redirect')
      })

      it('renders div instead of Redirect if user is not authed', () => {
        props.authedUser.isAuthenticated = false
        expect(comp().first().name()).toBe('div')
      })
    })
  })

  describe('interaction', () => {
    describe('form submission', () => {})

    describe('name Field', () => {})

    describe('email Field', () => {})

    describe('password Field', () => {})

    describe('password confirmation Field', () => {})

    describe('submit Field', () => {})

    describe('Link component', () => {})
  })

  it('passes a snapshot test', () => {
    expect(comp()).toMatchSnapshot()
  })
})
