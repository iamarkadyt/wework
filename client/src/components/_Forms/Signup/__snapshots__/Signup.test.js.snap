// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signup passes a snapshot test 1`] = `
<div
  className="Signup-container"
>
  <form
    onSubmit={[Function]}
  >
    <field
      label="Your name:"
      name="name"
      onChange={[Function]}
      placeholder="John Doe"
      type="text"
      value=""
    />
    <field
      label="Your email:"
      name="email"
      onChange={[Function]}
      placeholder="your.name@mail.com"
      type="text"
      value=""
    />
    <field
      label="Password:"
      name="password"
      onChange={[Function]}
      placeholder="Min. 6 characters"
      type="password"
      value=""
    />
    <field
      label="Confirm password:"
      name="passwordConfirm"
      onChange={[Function]}
      placeholder="Type in your password again"
      type="password"
      value=""
    />
    <br />
    <span>
      By creating an account, you agree to Conditions of Use and Privacy Notice of this Website
    </span>
    <div
      className="Signup-buttons"
    >
      <span
        className="Signup-options-info"
      >
        Already have an account? 
        <Link
          replace={false}
          to="/login"
        >
          Log In
        </Link>
      </span>
      <field
        containerStyle={
          Object {
            "margin": 0,
          }
        }
        inline={true}
        label="Sign Up"
        name="submit"
        type="submit"
      />
    </div>
  </form>
</div>
`;
